src/pru-motion-queue_test.cc

```diff
@@ -44,188 +44,122 @@ class MockPRUInterface : public PruHardwareInterface {
+      mmap->ring_buffer[execution_index_].state = STATE_FILLED;
+    mmap->status.index = execution_index_;
+    mmap->status.counter = loops_left;
   }
 
 private:
   struct MockPRUCommunication *mmap;
+  unsigned int execution_index_;
 };
 
-// Check that on init, the initial position is 0.
-TEST(RealtimePosition, init_pos) {
-  MotorsRegister absolute_pos_loops;
-  MockPRUInterface *pru_interface = new MockPRUInterface();
-  HardwareMapping *hmap = new HardwareMapping();
-  PRUMotionQueue motion_backend(hmap, (PruHardwareInterface*) pru_interface);
-
-  motion_backend.GetMotorsLoops(&absolute_pos_loops);
-
-  const MotorsRegister expected = {0, 0, 0, 0, 0, 0, 0, 0};
-  EXPECT_THAT(expected, ::testing::ContainerEq(absolute_pos_loops));
+static void check_buffer_value(PRUMotionQueue *motion_backend,
+                               const unsigned size) {
+  const unsigned int buffer = motion_backend->GetPendingElements(NULL);
+  EXPECT_EQ(buffer, size);
+}
 
-  delete pru_interface;
-  delete hmap;
+TEST(PruMotionQueue, status_init) {
+  MotorsRegister absolute_pos_loops;
+  MockPRUInterface pru_interface = MockPRUInterface();
+  HardwareMapping hmap = HardwareMapping();
+  PRUMotionQueue motion_backend(&hmap, (PruHardwareInterface*) &pru_interface);
+  check_buffer_value(&motion_backend, 0);
 }
 
-// Check that SimRun(1, 0) produce the same absolute position as
-// SimRun(2, MAX_SEGMENT_LOOPS).
-TEST(RealtimePosition, zero_loops_edge) {
-  //motor 0 +, motor 1 -, motor 2 -, 150 loops. fractions: /1 /3 /5
-  static struct MotionSegment segment = {
-    STATE_FILLED /*state*/, 0x00 | 1 << 1 | 1 << 2 /*direction bits*/,
-    0 /*loops accel*/, 150u /*loops travel*/, 0 /*loops decel*/,
-    0 /*aux*/, 0 /*accel_series_index*/, 0 /*hires_accel_cycles*/,
-    0 /*travel_delay_cycles*/,
-    { 0xFFFFFFFF, 0x55555555, 0x33333333, 0, 0, 0, 0, 0}/*fractions*/,
-  };
+TEST(PruMotionQueue, single_exec) {
   MotorsRegister absolute_pos_loops;
-  MockPRUInterface *pru_interface = new MockPRUInterface();
-  HardwareMapping *hmap = new HardwareMapping();
-  PRUMotionQueue motion_backend(hmap, (PruHardwareInterface*) pru_interface);
+  MockPRUInterface pru_interface = MockPRUInterface();
+  HardwareMapping hmap = HardwareMapping();
+  PRUMotionQueue motion_backend(&hmap, (PruHardwareInterface*) &pru_interface);
 
-  motion_backend.Enqueue(&segment);
+  static struct MotionSegment segment = {0};
   segment.state = STATE_FILLED;
   motion_backend.Enqueue(&segment);
-  pru_interface->SimRun(0, 0);
-
-  motion_backend.GetMotorsLoops(&absolute_pos_loops);
-
-  const MotorsRegister expected = {150, -50, -30, 0, 0, 0, 0, 0};
-  EXPECT_THAT(expected, ::testing::ContainerEq(absolute_pos_loops));
-
-  pru_interface->SimRun(1, 150);
-  motion_backend.GetMotorsLoops(&absolute_pos_loops);
-
-  EXPECT_THAT(expected, ::testing::ContainerEq(absolute_pos_loops));
-
-  delete pru_interface;
-  delete hmap;
+  pru_interface.SimRun(1, 0);
+  check_buffer_value(&motion_backend, 1);
```

Comment:
exactly, now we evaluate that part in motor-operations.cc so it has been moved in motor-operations_test